### PR TITLE
Fix some web requests being run after disposal of their owner

### LINKED ISSUE #3353: PaginatedScoreContainer adding drawables while being disposed
1. Open up profile overlay
2. Close profile overlay

`PaginatedScoreContainer` performs a web request and adds items when it succeeds, but it's possible for it to be disposed when this addition happens.

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/63007621-paginatedscorecontainer-adding-drawables-while-being-disposed?utm_campaign=plugin&utm_content=tracker%2F4707269&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F4707269&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by peppy**
Closes #3353.

### CODE REVIEW

File: osu.Game/Overlays/UserProfileOverlay.cs [hunk 1]
```diff
@@ -73,16 +73,15 @@ protected override void PopOut()
...
-
-            ShowUser(new User { Id = userId });
-        }
+        public void ShowUser(long userId) => ShowUser(new User { Id = userId });
 
         public void ShowUser(User user, bool fetchOnline = true)
         {
+            Show();
+
+            if (user.Id == Header?.User.Id)
```
Comment (peppy) [inline osu.Game/Overlays/UserProfileOverlay.cs]: The null check here should also resolve https://sentry.io/share/issue/0de998bd00b2417fa61f78f9c0cc47ed/

File: osu.Game/Overlays/BeatmapSet/Scores/ScoresContainer.cs [hunk 2]
```diff
@@ -62,7 +62,7 @@ public BeatmapInfo Beatmap
                 loading = true;
 
                 getScoresRequest = new GetScoresRequest(beatmap, beatmap.Ruleset);
-                getScoresRequest.Success += r => Scores = r.Scores;
+                getScoresRequest.Success += r => Schedule(() => Scores = r.Scores);
```
Comment (smoogipoo) [inline osu.Game/Overlays/BeatmapSet/Scores/ScoresContainer.cs]: Purposefully not disposed?

Comment (peppy) [reply osu.Game/Overlays/BeatmapSet/Scores/ScoresContainer.cs]: What do you mean by disposed? cancelled?

Comment (smoogipoo) [reply osu.Game/Overlays/BeatmapSet/Scores/ScoresContainer.cs]: Yeah

Comment (peppy) [reply osu.Game/Overlays/BeatmapSet/Scores/ScoresContainer.cs]: It gets cancelled when new scores are set on it. The `ScoresContainer` has a lifetime of forever.

we could cancel in dispose as a safety measure, won't really affect anything.

**Review by smoogipoo — APPROVED** [review]

**Pull request merged**